Docker/Dockerfile

```diff
@@ -0,0 +1,53 @@
+FROM lukemathwalker/cargo-chef:latest-rust-1.60.0-slim as builder
+
+# Dictates whether the underlying node should be built as a dev or
+# production binary. If empty, then the development mode will be built.
+# If passed as BUILD_MODE=--release, then the production will be built.
+ARG BUILD_MODE=""
+
+RUN echo "Will build image with mode: ${BUILD_MODE}"
+
+# Install basic dependencies
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get install -y --no-install-recommends \
+    tzdata \
+    git \
+    ca-certificates \
+    curl \
+    build-essential \
+    libssl-dev \
+    pkg-config \
+    clang \
+    cmake \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install the fmt
+RUN rustup component add rustfmt
+
+# Set the workdir to the build folder
+WORKDIR "$WORKDIR/build/narwhal"
+
+# Copy all the files in the workdir excluding everything
+# from the .dockerignore
+COPY . .
```

Comment:
That's a very good point. Initially I had the `Dockefile` on the project's root directory - the usual approach - but especially with the increasing number of files to help on the setup (e.x `authorities` folder, `entry.sh` file etc) I considered moving this on a separate dedicated folder which obviously leads to those edge cases (e.x building via `docker build` directly rather than using docker-compose). 

So for now I'll do two things
1. `open an issue` as you suggested
2. `include on the README file` instructions of how to build the image without docker-compose, via the command `docker build -f Dockerfile ../` , which will basically allow someone to define the context path on the root folder and use the Dockerfile to build the image